### PR TITLE
Proof of concept for condition unit tests with proper annotation discovery

### PR DESCRIPTION
**Pull request by fubhy**

### CODE REVIEW

File: tests/src/Condition/NodeIsPublishedTest.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  /**
+   * Tests the context definitions.
+   *
+   * @covers ::contextDefinitions()
+   */
+  public function testContextDefinition() {
+    // Test that the 'path' context is properly defined.
+    $context = $this->condition->getContext('node');
+    $this->assertInstanceOf('Drupal\rules\Context\ContextInterface', $context);
```
Comment (fago) [inline tests/src/Condition/NodeIsPublishedTest.php:75]: Those checks seem to be rather verbose. Maybe we can provide a helper that does general context assertions given a context key.

Comment (fubhy) [inline tests/src/Condition/NodeIsPublishedTest.php:75]: Sure. This is not the important part of this PR :P. This is a proof of concept for the discovery!

File: tests/src/Mock/ContainerFactory.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+namespace Drupal\rules\Tests\Mock;
+
+use Drupal\Core\Plugin\Factory\ContainerFactory as CoreContainerFactory;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Mock container factory for testing plugins.
```
Comment (fago) [inline tests/src/Mock/ContainerFactory.php:15]: Why do we need this. We should clearly document for each mock class why it's needed.

**Comment by fubhy** [comment]
I don't think that this is a good idea anymore. It's mixing integration and unit test and there is a high probability for it breaking if anything else in a given Drupal setup modifies anything with the annotation or adds custom stuff.

**Pull request closed without merging**